File: backend/account/src/main/java/com/example/account/model/Customer.java
package com.example.account.model;

import jakarta.persistence.Entity;

@Entity
public class Customer extends AuditedEntity {

  private String name;

  private String surname;

  public Customer() {}

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getSurname() {
    return surname;
  }

  public void setSurname(String surname) {
    this.surname = surname;
  }
}

File: frontend/src/app/home/customer.ts
import { Account } from '../accounts/account';

export class Customer {
  public static deserialise(json: any): Customer {
    return new Customer(
      json.id,
      json.name,
      json.surname,
      Array.isArray(json.accounts)
        ? json.accounts.map(Account.deserialise)
        : [],
    );
  }

  constructor(
    public id: string,
    public name: string,
    public surname: string,
    public accounts: Account[],
  ) {}
}
